src/skmultiflow/trees/arf_hoeffding_tree.py

```diff
@@ -286,21 +109,28 @@ def __init__(self,
         self._random_state = check_random_state(self.random_state)
 
     def _new_learning_node(self, initial_class_observations=None):
-        """Create a new learning node. The type of learning node depends on the tree configuration."""
+        """Create a new learning node. The type of learning node depends on the
+        tree configuration."""
         if initial_class_observations is None:
             initial_class_observations = {}
         # MAJORITY CLASS
         if self._leaf_prediction == MAJORITY_CLASS:
-            return self.RandomLearningNode(initial_class_observations, self.max_features,
-                                           random_state=self._random_state)
+            return ARFLearningNode(
```

Comment:
The main purpose was trying to find a way to tie the node implementation with the trees that use them.

As I previously mentioned:

> I've tried to standardize the names of the new files as much as I could. I had to rename some classes to comply with the name of the trees they belong to.

> Nevertheless, there might be still some inconsistencies in the names. Unlike the tree algorithms (Issue #138), perhaps we could standardize the leaves names in this PR, since the users do not access them directly via the API.

Maybe we could use something like `RandomLearningNodeClassification` or some other variation that might make clearer the node's purpose. Comments?

Reply:
This is indeed a tricky topic. Using the names of the trees could be confusing if we want to re-use them in different methods (e.g. `ActiveLearningNode` from the Hoeffding Tree). I like the idea of stating the type of learning (classification or regression). `RandomLearningNodeClassification` seems like a good compromise.

Comments @hmgomes?

Reply:
I have renamed the `ARFLearningNode*` classes to `RandomLearningNode*`.